torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -1510,6 +1510,14 @@ XLAGraphExecutor::SyncTensorsGraphInternal(
           coll.hash, torch::lazy::Hash(buffer_donor_index));
     }
   }
+  if (ShardingUtil::GetAutoSharding()) {
+    std::string xla_auto_spmd = "XLA_AUTO_SPMD=1";
+    coll.hash = torch::lazy::HashCombine(
+        coll.hash, torch::lazy::StringHash(xla_auto_spmd.c_str()));
```

Comment:
I think you can just 
```
    coll.hash = torch::lazy::HashCombine(
        coll.hash, torch::lazy::MHash(ShardingUtil::GetAutoSharding()));
    coll.hash = torch::lazy::HashCombine(
        coll.hash, torch::lazy::StringHash(sys_util::GetEnvString("XLA_AUTO_SPMD_MESH", "").c_str()));
```
The order of the hash is fixed, we don't need to prefix them with string.

Reply:
given that we have to hash each IR already, adding 2 small hash won't hurt the non-auto-sharding performance.

Reply:
Sounds good.